examples/argv.js

```diff
@@ -0,0 +1,100 @@
+"use strict";
+
+// Run me with Node to see my output!
+// $ node examples/argv.js foo -x
+// $ node examples/argv.js -x=1 bar
+// $ node examples/argv.js --some --thing=x
+
+let util = require("util");
+let P = require("..");
+
+let CLI = P.createLanguage({
+  expression: function(r) {
+    // whitespace-separated words, strings and options
+    return P.alt(r.word, r.string, r.option)
+      .sepBy(P.whitespace)
+      .trim(P.optWhitespace);
+  },
+
+  option: function(r) {
+    // one of possible quotes, then sequence of anything except that quote (unless escaped), then the same quote
+    return P.seq(
+      P.alt(P.string("-").then(P.regex(/[a-z]/)), P.string("--").then(r.word)),
+      P.alt(P.string("=").then(r.word), P.of(true))
+    );
+  },
+
+  word: function() {
+    // 1 char of anything except forbidden symbols and dash, then 0+ chars of anything except forbidden symbols
+    return P.regex(/[^-=\s"'][^=\s"']*/);
+  },
+
+  string: function() {
+    // one of possible quotes, then sequence of anything except that quote (unless escaped), then the same quote
+    return P.oneOf(`"'`).chain(function(q) {
+      return P.alt(
+        P.regex(new RegExp(`[^\\\\${q}]+`)),
+        P.string(`\\`).then(P.any)
+      )
+        .many()
+        .tie()
+        .skip(P.string(q));
+    });
+  }
+});
+
+let args = process.argv.slice(2).join(" ");
```

Comment:
I think it might make more sense to take the entire options string as `argv[2]` like `node cli.js "--foo --qux=guz -b"` to show that this is parsing out one whole string and not relying on the shell (bash, etc) to parse for us.